File: ios/Classes/MethodNames.swift
//
//  MethodNames.swift
//  flutter_cast_framework
//
//  Created by Gianluca Paradiso on 09/11/2019.
//

import Foundation

enum MethodNames : String {
    case onCastStateChanged = "CastContext.onCastStateChanged"
    case showCastDialog = "showCastDialog"

    // region SessionManager
    case onSessionStarting = "SessionManager.onSessionStarting"
    case onSessionStarted = "SessionManager.onSessionStarted"
    case onSessionStartFailed = "SessionManager.onSessionStartFailed"
    case onSessionEnding = "SessionManager.onSessionEnding"
    case onSessionEnded = "SessionManager.onSessionEnded"
    case onSessionResuming = "SessionManager.onSessionResuming"
    case onSessionResumed = "SessionManager.onSessionResumed"
    case onSessionResumeFailed = "SessionManager.onSessionResumeFailed"
    case onSessionSuspended = "SessionManager.onSessionSuspended"
    // end-region

    case getSessionMessageNamespaces = "CastSession.getSessionMessageNamespaces"
    case onMessageReceived = "CastSession.onMessageReceived"
}

File: android/src/main/kotlin/com/gianlucaparadise/flutter_cast_framework/MethodNames.kt
package com.gianlucaparadise.flutter_cast_framework

object MethodNames {
    const val onCastStateChanged = "CastContext.onCastStateChanged"
    const val showCastDialog = "showCastDialog"

    // region SessionManager
    const val onSessionStarting = "SessionManager.onSessionStarting"
    const val onSessionStarted = "SessionManager.onSessionStarted"
    const val onSessionStartFailed = "SessionManager.onSessionStartFailed"
    const val onSessionEnding = "SessionManager.onSessionEnding"
    const val onSessionEnded = "SessionManager.onSessionEnded"
    const val onSessionResuming = "SessionManager.onSessionResuming"
    const val onSessionResumed = "SessionManager.onSessionResumed"
    const val onSessionResumeFailed = "SessionManager.onSessionResumeFailed"
    const val onSessionSuspended = "SessionManager.onSessionSuspended"
    // end-region

    const val getSessionMessageNamespaces = "CastSession.getSessionMessageNamespaces"
    const val onMessageReceived = "CastSession.onMessageReceived"
}